connect-inject/container_init.go

```diff
@@ -298,31 +310,45 @@ cat <<EOF >/consul/connect-inject/service-defaults.hcl
 kind = "service-defaults"
 name = "{{ .ServiceName }}"
 protocol = "{{ .ServiceProtocol }}"
+{{- if .ConsulNamespace }}
+namespace = "{{ .ConsulNamespace }}"
+{{- end }}
 EOF
 {{- end }}
+
 {{- if .AuthMethod }}
 /bin/consul login -method="{{ .AuthMethod }}" \
   -bearer-token-file="/var/run/secrets/kubernetes.io/serviceaccount/token" \
   -token-sink-file="/consul/connect-inject/acl-token" \
+  {{- if .ConsulNamespace }}
+  -namespace="{{ .ConsulNamespace }}" \
+  {{- end }}
   -meta="pod=${POD_NAMESPACE}/${POD_NAME}"
 {{- /* The acl token file needs to be read by the lifecycle-sidecar which runs
        as non-root user consul-k8s. */}}
 chmod 444 /consul/connect-inject/acl-token
 {{- end }}
+
 {{- if .WriteServiceDefaults }}
 {{- /* We use -cas and -modify-index 0 so that if a service-defaults config
        already exists for this service, we don't override it */}}
 /bin/consul config write -cas -modify-index 0 \
   {{- if .AuthMethod }}
   -token-file="/consul/connect-inject/acl-token" \
   {{- end }}
+  {{- if .ConsulNamespace }}
+  -namespace="{{ .ConsulNamespace }}" \
+  {{- end }}
   /consul/connect-inject/service-defaults.hcl || true
 {{- end }}
 
 /bin/consul services register \
   {{- if .AuthMethod }}
   -token-file="/consul/connect-inject/acl-token" \
   {{- end }}
+  {{- if .ConsulNamespace }}
+  -namespace={{ .ConsulNamespace }} \
+  {{- end }}
```

Comment:
Also noticed the other day that `/bin/consul connect envoy` command needs the namespace flag (GH doesn't let me comment on the code that wasn't changed). One thing we can do is just provide the `CONSUL_NAMESPACE` env variable to the container and remove the flag from all commands to simplify it.